src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala

```diff
@@ -475,36 +475,15 @@ object DepositHandler {
     afterBaseUrl.substring(afterBaseUrl.indexOf("/data/") + 1)
   }
 
-  case class MakeAllGroupWritable(permissions: String, id: String) extends SimpleFileVisitor[Path] {
-    override def visitFile(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
-      log.debug(s"[$id] Setting the following permissions $permissions on file $path")
-      try {
-        Files.setPosixFilePermissions(path, PosixFilePermissions.fromString(permissions))
-        FileVisitResult.CONTINUE
-      } catch {
-        case usoe: UnsupportedOperationException => log.error("Not on a POSIX supported file system"); FileVisitResult.TERMINATE
-        case cce: ClassCastException => log.error("No file permission elements in set"); FileVisitResult.TERMINATE
-        case ioe: IOException => log.error(s"Could not set file permissions on $path"); FileVisitResult.TERMINATE
-        case se: SecurityException => log.error(s"Not enough privileges to set file permissions on $path"); FileVisitResult.TERMINATE
-      }
-    }
-
-    override def postVisitDirectory(dir: Path, ex: IOException): FileVisitResult = {
-      log.debug(s"[$id] Setting the following permissions $permissions on directory $dir")
-      Files.setPosixFilePermissions(dir, PosixFilePermissions.fromString(permissions))
-      if (ex == null) FileVisitResult.CONTINUE
-      else FileVisitResult.TERMINATE
-    }
-  }
-
   def isOnPosixFileSystem(file: File): Boolean = Try(Files.getPosixFilePermissions(file.toPath)).fold(_ => false, _ => true)
 
   def moveBagToStorage(depositDir: File, storageDir: File)(implicit settings: Settings, id: String): Try[File] =
     Try {
       log.debug(s"[$id] Moving bag to permanent storage")
-      if (isOnPosixFileSystem(depositDir))
-        Files.walkFileTree(depositDir.toPath, MakeAllGroupWritable(settings.depositPermissions, id))
-      Files.move(depositDir.toPath.toAbsolutePath, storageDir.toPath.toAbsolutePath).toFile
+      val tempDir = new File(settings.tempDir, id)
+      val storageDir = new File(settings.depositRootDir, id)
+      FilesPermissionService.changePermissionsForDirectoryAndContent(tempDir, settings.depositPermissions, id).get
```

Comment:
try to write this `Try` block as a for-comprehension. That way you loose the `.get`.